### PR TITLE
Update Rules

### PR DESCRIPTION
**Pull request by lishaduck**
Some opinionated changes I made. Feel free to close this if you disagree.

* Label configs for debugging in the ESLint Config Inspector.
* Switch off deprecated rules.
* Enable most of the recommended Unicorn rules. (Most controversial commit)

### CODE REVIEW

**Comment by lishaduck** [comment]
Docusaurus's ESM support is a sham![^1] They're just using Jiti, so this PR `require`s... the dreaded `eslint-disable` comment: unjs/jiti#101

[^1]: Very much an exaggeration, full of hyperbole: Jiti is certainly ESM, they're just not ESM, we have to be... patient! 😜

**Review by lishaduck — COMMENTED** [review]
Looking back on this, this has some good stuff that could be extracted, but is based on #161, and is heavily conflicted. On top of that, there are definitely some things wrong wtith this approch. I might revist someday, so I'm not closing; there's some context for my future self.

**Comment by AndreaPontrandolfo** [comment]
@lishaduck coming back to this, i think we ended up integrating most of the changes here within https://github.com/AndreaPontrandolfo/sheriff/pull/326.

About the unicorn changes, it's definitely a change I don't want. I'm pretty happy with how we are handling unicorn rules right now, i want to keep hand-picking them because some of them are extremely opinionated, I don't want to introduce any rule without a specific reason, and some of those rules just don't fit within Sheriff philosophy.

About the config labels, i don't see much of a usecase for them right now, but i'm **not** opposed about introducing them.

About the deprecated rules, i left a in-code comment.

File: packages/eslint-config-sheriff/src/deprecatedOverrides.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import type { TSESLint } from '@typescript-eslint/utils';
+
+export const deprecatedRecommendedOverrides = {
```
[the remark this answers is not in the document]
Comment (lishaduck) [reply packages/eslint-config-sheriff/src/deprecatedOverrides.ts:3]: They _were_ in the recommended configs that the config extended.

**Comment by lishaduck** [comment]
> About the unicorn changes, it's definitely a change I don't want. I'm pretty happy with how we are handling unicorn rules right now, i want to keep hand-picking them because some of them are extremely opinionated, I don't want to introduce any rule without a specific reason, and some of those rules just don't fit within Sheriff philosophy.

Yeah, I agree there. I do want to revisit some of them, but I agree we shouldn't extend their recommended config.

> About the config labels, i don't see much of a usecase for them right now, but i'm **not** opposed about introducing them.

They're primarily for the [ESLint config inspector](https://github.com/eslint/config-inspector), but are generally recommended (the inspector really helps with debugging). I will say that the prefered format is `eslint-config-sheriff/someconfigpartname`, which I didn't know at the time.

**Comment by lishaduck** [comment]
Superseded by #331.

**Pull request closed without merging**